cypress/integration/preferences/notifications.ts

```diff
@@ -24,17 +24,65 @@ describe("global subscription settings", () => {
 });
 
 describe("user subscriptions table", () => {
-  it("shows all of a user's subscriptions and expands with details", () => {
+  beforeEach(() => {
     cy.visit(pageRoute);
+  });
+
+  it("shows all of a user's subscriptions and expands with details", () => {
     cy.dataCy("subscription-row").should("have.length", 3);
 
     cy.dataCy("regex-selectors").should("not.be.visible");
     cy.dataCy("trigger-data").should("not.be.visible");
-    cy.get("tr button").first().click();
+    cy.dataCy("subscription-row")
+      .eq(0)
+      .within(() => {
+        cy.get("button").first().click();
+      });
     cy.dataCy("regex-selectors").should("be.visible");
     cy.dataCy("trigger-data").should("not.be.visible");
-    cy.get("tr button").last().click();
+    cy.dataCy("subscription-row")
+      .eq(2)
+      .within(() => {
+        cy.get("button").first().click();
+      });
     cy.dataCy("regex-selectors").should("be.visible");
     cy.dataCy("trigger-data").should("be.visible");
   });
+
+  it("Shows the selected count in the 'Delete' button", () => {
+    cy.dataCy("subscription-row")
+      .eq(0)
+      .within(() => {
+        cy.get("input[type=checkbox]").check({ force: true });
+      });
+    cy.dataCy("delete-some-button").contains("Delete (1)");
+
+    cy.get("thead").within(() => {
+      cy.get("input[type=checkbox]").check({ force: true });
+    });
+    cy.dataCy("delete-some-button").contains("Delete (3)");
+
+    cy.get("thead").within(() => {
+      cy.get("input[type=checkbox]").uncheck({ force: true });
+    });
+    cy.dataCy("delete-some-button").contains("Delete");
+    cy.dataCy("delete-some-button").should(
+      "have.attr",
+      "aria-disabled",
+      "true"
+    );
+  });
+
+  describe("Deleting subscriptions", { testIsolation: false }, () => {
```

Comment:
[suggestion] for test isolation purposes, perhaps we can add a subscription in this test and then delete it here. certainly not required though, just trying to get ahead of the game :)

Reply:
This `beforeEach`already exists at the parent level so it's applied here too 😎

I was definitely thinking along the lines of what Arjun said and counting on resetting the DB here. Given that adding a subscription is a fairly different task that involves navigating to a new page, and is unrelated functionally, I thought it may make the scope of this test too big.